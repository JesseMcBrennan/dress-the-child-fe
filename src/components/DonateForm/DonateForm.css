.donate-form {
  width: 20rem;
  margin: auto; }
