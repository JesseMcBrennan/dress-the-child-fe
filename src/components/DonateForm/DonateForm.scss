$teal: #047177;

.donate-form {
  width: 20rem;
  margin: 1rem auto;
}

.cc-input {
  width: 293px;
  margin: 0.5rem 0;
  font-size: 0.8rem;
  background-color: white;
  padding: 10px 12px;
  border-radius: 4px;
  border: 1px solid transparent;
  box-shadow: 0 1px 3px 0 #e6ebf1;
  -webkit-transition: box-shadow 150ms ease;
  transition: box-shadow 150ms ease;
}

.cc-input:disabled {
  background-color: lightgray;
}
.first-name-cc {
  margin: 0 0 0.5rem 0;
}

.StripeElement {
  margin: 0.5rem 0;
  width: auto;
  background-color: white;
  padding: 10px 12px;
  border-radius: 4px;
  border: 1px solid transparent;
  box-shadow: 0 1px 3px 0 #e6ebf1;
  -webkit-transition: box-shadow 150ms ease;
  transition: box-shadow 150ms ease;
}

.StripeElement--focus {
  box-shadow: 0 1px 3px 0 #cfd7df;
}

.StripeElement--invalid {
  border-color: #fa755a;
}

.StripeElement--webkit-autofill {
  background-color: #fefde5 !important;
}

.submit-btn {
  color: $teal;
  padding: 0.3rem 1.3rem;
  border: 0.1rem solid $teal;
  font-size: 1.7rem;
  text-decoration: none;
  border-radius: 0.15rem;
  cursor: pointer;
  background: whitesmoke;
  margin-top: -20px;
}

.submit-btn:hover {
  color: whitesmoke;
  border: 0.1rem solid whitesmoke;
  background: $teal;
}

.submit-btn:active {
  color: $teal;
  border: 0.1rem solid $teal;
  background: whitesmoke;
}

.email-msg {
  font-size: 0.9rem;
  color: black;
  margin: 0;
}

.processing-msg {
  font-size: 0.9rem;
  width: auto;
  margin: 0.1rem 0;
  color: black;
}

.donation-msg {
  font-size: 1rem;
}

.transaction-msg {
  font-size: 0.9rem;
  width: auto;
  margin: 0.1rem 0;
}

.final-donation-msg {
  font-size: 1rem;
  width: auto;
  margin: 0.1rem 0;
}

.final-total {
  font-size: 2rem;
  font-weight: 300;
  margin: 0 0 1.3rem 0;
}

.first-name {
  text-transform: capitalize;
}

.loading-gif {
  margin-top: 2rem;
  background: no-repeat center/200px url('../../images/loading-gif.gif');
  width: 400px;
  height: 400px;
  margin: auto;
}

.thanks-msg {
  font-weight: 300;
}

.error-msg {
  font-weight: 300;
}

.anonymous-text {
  margin: 0 0 0.1rem 0;
  font-size: 0.9rem;
}

.anonymous-section {
  display: flex;
  flex-direction: row;
  width: 20rem;
  justify-content: center;
  align-items: center;
  height: auto;
  width: auto;
}
