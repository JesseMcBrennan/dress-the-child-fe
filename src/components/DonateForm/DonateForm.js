import React, { Component } from 'react';
import { CardElement, injectStripe } from 'react-stripe-elements';

import './DonateForm.css';

export class DonateForm extends Component {
  constructor(props) {
    super(props);
    this.state = {
      firstName: '',
      lastName: '',
      amount: '',
      phone: '',
      email: '',
      city: '',
      state: '',
      amountTotal: 0,
      fees: 0,
      disableBtn: false,
      complete: false,
      isLoading: false,
      isError: false,
      isChecked: false
    };
  }

  handleChange = e => {
    const { name, value } = e.target;
    this.setState({ [name]: value }, () => this.addTotal());
  };

  handleCheckbox = e => {
    if (this.state.isChecked) {
      this.setState({ isChecked: false });
    } else {
      this.setState({ isChecked: true });
    }
  };

  addTotal = () => {
    const amount = parseInt(this.state.amount);
    const amountTotal = (amount * 0.0233 + 0.3 + amount).toFixed(2);
    const fees = (amountTotal - amount).toFixed(2);
    this.setState({ amountTotal, fees });
  };

  submit = async e => {
    e.preventDefault();
    const { firstName, lastName, email, city, state, amountTotal } = this.state;
    let { token } = await this.props.stripe.createToken({
      name: `${firstName} ${lastName}`,
      address_city: `${city}`,
      address_state: `${state}`,
      address_line2: `${email}`
    });
    this.setState({ disableBtn: true, isLoading: true });
    this.makeCharge(token, amountTotal);
  };

  makeCharge = async (token, amountTotal) => {
    try {
      let response = await fetch(
        'https://dress-the-child-be.herokuapp.com/api/v1/charges/',
        {
          method: 'POST',
          headers: { 'Content-Type': 'application/json' },
          body: JSON.stringify({
            stripeToken: token.id,
            stripeAmount: amountTotal * 100
          })
        }
      );
      if (response.ok) {
        this.setState({ complete: true, disableBtn: false, isLoading: false });
      } else {
        this.setState({ isError: true, isLoading: false });
      }
    } catch {
      this.setState({ isError: true });
    }
  };

  render() {
    const style = {
      base: {
        color: '#32325d',
        lineHeight: '18px',
        fontFamily: '"Helvetica Neue", Helvetica, sans-serif',
        fontSmoothing: 'antialiased',
        fontSize: '16px',
        '::placeholder': {
          color: '#aab7c4'
        }
      },
      invalid: {
        color: '#fa755a',
        iconColor: '#fa755a'
      }
    };

    const {
      complete,
      firstName,
      lastname,
      email,
      city,
      state,
      amount,
      fees,
      amountTotal,
      isLoading,
      isChecked,
      isError
    } = this.state;
    if (isLoading) {
      return <div className="loading-gif" />;
    }

    if (complete) {
      return (
        <h1 className="thanks-msg">
          Thanks <span className="first-name">{firstName}</span>! <br />
          Your donation of ${amountTotal} has been processed. <br />
          100% of your donation will help a child in need.
        </h1>
      );
    } else if (isError) {
      return (
        <h1 className="error-msg">
          There was an error processing your payment
        </h1>
      );
    }

    return (
      <div className="donate-form">
        <div className="anonymous-section">
          <p className="anonymous-text">To donate anonymously click here: </p>
          <input
            type="checkbox"
            className="anonymous-checkbox"
            onChange={this.handleCheckbox}
          />
        </div>
        <form onSubmit={e => this.submit(e)}>
          <input
            type="text"
            placeholder="First Name"
            value={firstName}
            name="firstName"
            onChange={this.handleChange}
            className="cc-input first-name-cc"
            disabled={isChecked && true}
          />
          <input
            type="text"
            placeholder="Last Name"
            value={lastname}
            name="lastName"
            onChange={this.handleChange}
            className="cc-input"
            disabled={isChecked && true}
          />
          <input
            type="text"
            placeholder="Email"
            value={email}
            name="email"
            onChange={this.handleChange}
            className="cc-input"
          />
          <p className="email-msg">
            Your email address will only be used for receipt
          </p>
          <input
            type="text"
            placeholder="Amount"
            value={amount}
            name="amount"
            onChange={e => this.handleChange(e)}
            className="cc-input"
          />
          <p className="processing-msg">
            Plus an additional ${fees > 0 ? fees : 0} to cover processing fees
          </p>
          <input
            type="text"
            placeholder="City"
            value={city}
            name="city"
            onChange={this.handleChange}
            className="cc-input"
            disabled={isChecked && true}
          />
          <input
            type="text"
            placeholder="State"
            value={state}
            name="state"
            onChange={this.handleChange}
            className="cc-input"
            disabled={isChecked && true}
          />
          <CardElement style={style} />
          <div className="final-donation-msg">
            <p className="transaction-msg">
              Transactions are secure and encrypted
            </p>
          </div>
          <h3 className="final-total">
            Total: {amountTotal > 0 ? amountTotal : 0}
          </h3>
          <button
            className="submit-btn"
            disabled={this.state.disableBtn ? true : false}
          >
            Confirm
          </button>
        </form>
      </div>
    );
  }
}

export default injectStripe(DonateForm);
