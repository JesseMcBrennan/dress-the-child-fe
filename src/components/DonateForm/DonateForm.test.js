import React from 'react';
import { shallow } from 'enzyme';
import { DonateForm } from './DonateForm';

describe('MissionPage', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<DonateForm />);
  });
  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});

describe('handleChange', () => {
  it('should update state', () => {
    let wrapper = shallow(<handleChange />);
    const mockFirstNameEvent = {
      target: { name: 'firstName', value: 'Kurtious' }
    };
    const mockLastNameEvent = {
      target: { name: 'lastName', value: 'Kurtains' }
    };
    const mockAmountEvent = { target: { name: 'amount', value: '600' } };
    const mockPhoneEvent = { target: { name: 'phone', value: '5128675309' } };
    const mockEmailEvent = {
      target: { name: 'email', value: 'kurtious.kurtains@gmail.com' }
    };
    const mockPasswordEvent = {
      target: { name: 'email', value: 'kurtious.kurtains@gmail.com' }
    };
  });
});
