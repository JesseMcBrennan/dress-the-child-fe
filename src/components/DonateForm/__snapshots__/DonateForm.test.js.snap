// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MissionPage should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DonateForm />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="anonymous-section"
        >
          <p
            className="anonymous-text"
          >
            To donate anonymously click here: 
          </p>
          <input
            className="anonymous-checkbox"
            onChange={[Function]}
            type="checkbox"
          />
        </div>,
        <form
          onSubmit={[Function]}
        >
          <input
            className="cc-input first-name-cc"
            disabled={false}
            name="firstName"
            onChange={[Function]}
            placeholder="First Name"
            type="text"
            value=""
          />
          <input
            className="cc-input"
            disabled={false}
            name="lastName"
            onChange={[Function]}
            placeholder="Last Name"
            type="text"
          />
          <input
            className="cc-input"
            name="email"
            onChange={[Function]}
            placeholder="Email"
            type="text"
            value=""
          />
          <p
            className="email-msg"
          >
            Your email address will only be used for receipt
          </p>
          <input
            className="cc-input"
            name="amount"
            onChange={[Function]}
            placeholder="Amount"
            type="text"
            value=""
          />
          <p
            className="processing-msg"
          >
            Plus an additional $
            0
             to cover processing fees
          </p>
          <input
            className="cc-input"
            disabled={false}
            name="city"
            onChange={[Function]}
            placeholder="City"
            type="text"
            value=""
          />
          <input
            className="cc-input"
            disabled={false}
            name="state"
            onChange={[Function]}
            placeholder="State"
            type="text"
            value=""
          />
          <CardElement
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onReady={[Function]}
            style={
              Object {
                "base": Object {
                  "::placeholder": Object {
                    "color": "#aab7c4",
                  },
                  "color": "#32325d",
                  "fontFamily": "\\"Helvetica Neue\\", Helvetica, sans-serif",
                  "fontSize": "16px",
                  "fontSmoothing": "antialiased",
                  "lineHeight": "18px",
                },
                "invalid": Object {
                  "color": "#fa755a",
                  "iconColor": "#fa755a",
                },
              }
            }
          />
          <div
            className="final-donation-msg"
          >
            <p
              className="transaction-msg"
            >
              Transactions are secure and encrypted
            </p>
          </div>
          <h3
            className="final-total"
          >
            Total: 
            0
          </h3>
          <button
            className="submit-btn"
            disabled={false}
          >
            Confirm
          </button>
        </form>,
      ],
      "className": "donate-form",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p
              className="anonymous-text"
            >
              To donate anonymously click here: 
            </p>,
            <input
              className="anonymous-checkbox"
              onChange={[Function]}
              type="checkbox"
            />,
          ],
          "className": "anonymous-section",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "To donate anonymously click here: ",
              "className": "anonymous-text",
            },
            "ref": null,
            "rendered": "To donate anonymously click here: ",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "anonymous-checkbox",
              "onChange": [Function],
              "type": "checkbox",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              className="cc-input first-name-cc"
              disabled={false}
              name="firstName"
              onChange={[Function]}
              placeholder="First Name"
              type="text"
              value=""
            />,
            <input
              className="cc-input"
              disabled={false}
              name="lastName"
              onChange={[Function]}
              placeholder="Last Name"
              type="text"
            />,
            <input
              className="cc-input"
              name="email"
              onChange={[Function]}
              placeholder="Email"
              type="text"
              value=""
            />,
            <p
              className="email-msg"
            >
              Your email address will only be used for receipt
            </p>,
            <input
              className="cc-input"
              name="amount"
              onChange={[Function]}
              placeholder="Amount"
              type="text"
              value=""
            />,
            <p
              className="processing-msg"
            >
              Plus an additional $
              0
               to cover processing fees
            </p>,
            <input
              className="cc-input"
              disabled={false}
              name="city"
              onChange={[Function]}
              placeholder="City"
              type="text"
              value=""
            />,
            <input
              className="cc-input"
              disabled={false}
              name="state"
              onChange={[Function]}
              placeholder="State"
              type="text"
              value=""
            />,
            <CardElement
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onReady={[Function]}
              style={
                Object {
                  "base": Object {
                    "::placeholder": Object {
                      "color": "#aab7c4",
                    },
                    "color": "#32325d",
                    "fontFamily": "\\"Helvetica Neue\\", Helvetica, sans-serif",
                    "fontSize": "16px",
                    "fontSmoothing": "antialiased",
                    "lineHeight": "18px",
                  },
                  "invalid": Object {
                    "color": "#fa755a",
                    "iconColor": "#fa755a",
                  },
                }
              }
            />,
            <div
              className="final-donation-msg"
            >
              <p
                className="transaction-msg"
              >
                Transactions are secure and encrypted
              </p>
            </div>,
            <h3
              className="final-total"
            >
              Total: 
              0
            </h3>,
            <button
              className="submit-btn"
              disabled={false}
            >
              Confirm
            </button>,
          ],
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "cc-input first-name-cc",
              "disabled": false,
              "name": "firstName",
              "onChange": [Function],
              "placeholder": "First Name",
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "cc-input",
              "disabled": false,
              "name": "lastName",
              "onChange": [Function],
              "placeholder": "Last Name",
              "type": "text",
              "value": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "cc-input",
              "name": "email",
              "onChange": [Function],
              "placeholder": "Email",
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Your email address will only be used for receipt",
              "className": "email-msg",
            },
            "ref": null,
            "rendered": "Your email address will only be used for receipt",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "cc-input",
              "name": "amount",
              "onChange": [Function],
              "placeholder": "Amount",
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Plus an additional $",
                0,
                " to cover processing fees",
              ],
              "className": "processing-msg",
            },
            "ref": null,
            "rendered": Array [
              "Plus an additional $",
              0,
              " to cover processing fees",
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "cc-input",
              "disabled": false,
              "name": "city",
              "onChange": [Function],
              "placeholder": "City",
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "cc-input",
              "disabled": false,
              "name": "state",
              "onChange": [Function],
              "placeholder": "State",
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "className": undefined,
              "id": undefined,
              "onBlur": [Function],
              "onChange": [Function],
              "onFocus": [Function],
              "onReady": [Function],
              "style": Object {
                "base": Object {
                  "::placeholder": Object {
                    "color": "#aab7c4",
                  },
                  "color": "#32325d",
                  "fontFamily": "\\"Helvetica Neue\\", Helvetica, sans-serif",
                  "fontSize": "16px",
                  "fontSmoothing": "antialiased",
                  "lineHeight": "18px",
                },
                "invalid": Object {
                  "color": "#fa755a",
                  "iconColor": "#fa755a",
                },
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <p
                className="transaction-msg"
              >
                Transactions are secure and encrypted
              </p>,
              "className": "final-donation-msg",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Transactions are secure and encrypted",
                "className": "transaction-msg",
              },
              "ref": null,
              "rendered": "Transactions are secure and encrypted",
              "type": "p",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Total: ",
                0,
              ],
              "className": "final-total",
            },
            "ref": null,
            "rendered": Array [
              "Total: ",
              0,
            ],
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Confirm",
              "className": "submit-btn",
              "disabled": false,
            },
            "ref": null,
            "rendered": "Confirm",
            "type": "button",
          },
        ],
        "type": "form",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="anonymous-section"
          >
            <p
              className="anonymous-text"
            >
              To donate anonymously click here: 
            </p>
            <input
              className="anonymous-checkbox"
              onChange={[Function]}
              type="checkbox"
            />
          </div>,
          <form
            onSubmit={[Function]}
          >
            <input
              className="cc-input first-name-cc"
              disabled={false}
              name="firstName"
              onChange={[Function]}
              placeholder="First Name"
              type="text"
              value=""
            />
            <input
              className="cc-input"
              disabled={false}
              name="lastName"
              onChange={[Function]}
              placeholder="Last Name"
              type="text"
            />
            <input
              className="cc-input"
              name="email"
              onChange={[Function]}
              placeholder="Email"
              type="text"
              value=""
            />
            <p
              className="email-msg"
            >
              Your email address will only be used for receipt
            </p>
            <input
              className="cc-input"
              name="amount"
              onChange={[Function]}
              placeholder="Amount"
              type="text"
              value=""
            />
            <p
              className="processing-msg"
            >
              Plus an additional $
              0
               to cover processing fees
            </p>
            <input
              className="cc-input"
              disabled={false}
              name="city"
              onChange={[Function]}
              placeholder="City"
              type="text"
              value=""
            />
            <input
              className="cc-input"
              disabled={false}
              name="state"
              onChange={[Function]}
              placeholder="State"
              type="text"
              value=""
            />
            <CardElement
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onReady={[Function]}
              style={
                Object {
                  "base": Object {
                    "::placeholder": Object {
                      "color": "#aab7c4",
                    },
                    "color": "#32325d",
                    "fontFamily": "\\"Helvetica Neue\\", Helvetica, sans-serif",
                    "fontSize": "16px",
                    "fontSmoothing": "antialiased",
                    "lineHeight": "18px",
                  },
                  "invalid": Object {
                    "color": "#fa755a",
                    "iconColor": "#fa755a",
                  },
                }
              }
            />
            <div
              className="final-donation-msg"
            >
              <p
                className="transaction-msg"
              >
                Transactions are secure and encrypted
              </p>
            </div>
            <h3
              className="final-total"
            >
              Total: 
              0
            </h3>
            <button
              className="submit-btn"
              disabled={false}
            >
              Confirm
            </button>
          </form>,
        ],
        "className": "donate-form",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p
                className="anonymous-text"
              >
                To donate anonymously click here: 
              </p>,
              <input
                className="anonymous-checkbox"
                onChange={[Function]}
                type="checkbox"
              />,
            ],
            "className": "anonymous-section",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "To donate anonymously click here: ",
                "className": "anonymous-text",
              },
              "ref": null,
              "rendered": "To donate anonymously click here: ",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "anonymous-checkbox",
                "onChange": [Function],
                "type": "checkbox",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                className="cc-input first-name-cc"
                disabled={false}
                name="firstName"
                onChange={[Function]}
                placeholder="First Name"
                type="text"
                value=""
              />,
              <input
                className="cc-input"
                disabled={false}
                name="lastName"
                onChange={[Function]}
                placeholder="Last Name"
                type="text"
              />,
              <input
                className="cc-input"
                name="email"
                onChange={[Function]}
                placeholder="Email"
                type="text"
                value=""
              />,
              <p
                className="email-msg"
              >
                Your email address will only be used for receipt
              </p>,
              <input
                className="cc-input"
                name="amount"
                onChange={[Function]}
                placeholder="Amount"
                type="text"
                value=""
              />,
              <p
                className="processing-msg"
              >
                Plus an additional $
                0
                 to cover processing fees
              </p>,
              <input
                className="cc-input"
                disabled={false}
                name="city"
                onChange={[Function]}
                placeholder="City"
                type="text"
                value=""
              />,
              <input
                className="cc-input"
                disabled={false}
                name="state"
                onChange={[Function]}
                placeholder="State"
                type="text"
                value=""
              />,
              <CardElement
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                onReady={[Function]}
                style={
                  Object {
                    "base": Object {
                      "::placeholder": Object {
                        "color": "#aab7c4",
                      },
                      "color": "#32325d",
                      "fontFamily": "\\"Helvetica Neue\\", Helvetica, sans-serif",
                      "fontSize": "16px",
                      "fontSmoothing": "antialiased",
                      "lineHeight": "18px",
                    },
                    "invalid": Object {
                      "color": "#fa755a",
                      "iconColor": "#fa755a",
                    },
                  }
                }
              />,
              <div
                className="final-donation-msg"
              >
                <p
                  className="transaction-msg"
                >
                  Transactions are secure and encrypted
                </p>
              </div>,
              <h3
                className="final-total"
              >
                Total: 
                0
              </h3>,
              <button
                className="submit-btn"
                disabled={false}
              >
                Confirm
              </button>,
            ],
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "cc-input first-name-cc",
                "disabled": false,
                "name": "firstName",
                "onChange": [Function],
                "placeholder": "First Name",
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "cc-input",
                "disabled": false,
                "name": "lastName",
                "onChange": [Function],
                "placeholder": "Last Name",
                "type": "text",
                "value": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "cc-input",
                "name": "email",
                "onChange": [Function],
                "placeholder": "Email",
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Your email address will only be used for receipt",
                "className": "email-msg",
              },
              "ref": null,
              "rendered": "Your email address will only be used for receipt",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "cc-input",
                "name": "amount",
                "onChange": [Function],
                "placeholder": "Amount",
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Plus an additional $",
                  0,
                  " to cover processing fees",
                ],
                "className": "processing-msg",
              },
              "ref": null,
              "rendered": Array [
                "Plus an additional $",
                0,
                " to cover processing fees",
              ],
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "cc-input",
                "disabled": false,
                "name": "city",
                "onChange": [Function],
                "placeholder": "City",
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "cc-input",
                "disabled": false,
                "name": "state",
                "onChange": [Function],
                "placeholder": "State",
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "className": undefined,
                "id": undefined,
                "onBlur": [Function],
                "onChange": [Function],
                "onFocus": [Function],
                "onReady": [Function],
                "style": Object {
                  "base": Object {
                    "::placeholder": Object {
                      "color": "#aab7c4",
                    },
                    "color": "#32325d",
                    "fontFamily": "\\"Helvetica Neue\\", Helvetica, sans-serif",
                    "fontSize": "16px",
                    "fontSmoothing": "antialiased",
                    "lineHeight": "18px",
                  },
                  "invalid": Object {
                    "color": "#fa755a",
                    "iconColor": "#fa755a",
                  },
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <p
                  className="transaction-msg"
                >
                  Transactions are secure and encrypted
                </p>,
                "className": "final-donation-msg",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Transactions are secure and encrypted",
                  "className": "transaction-msg",
                },
                "ref": null,
                "rendered": "Transactions are secure and encrypted",
                "type": "p",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Total: ",
                  0,
                ],
                "className": "final-total",
              },
              "ref": null,
              "rendered": Array [
                "Total: ",
                0,
              ],
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Confirm",
                "className": "submit-btn",
                "disabled": false,
              },
              "ref": null,
              "rendered": "Confirm",
              "type": "button",
            },
          ],
          "type": "form",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
